baybe/acquisition/_adapter.py

```diff
@@ -1,6 +1,7 @@
 """Adapter for making BoTorch's acquisition functions work with BayBE models."""
 
-from typing import Any, Callable, Optional
+from collections.abc import Callable
```

Comment:
Yes, as far as I know that's the same reasoning for why one can now write `tuple[int]` instead of having to go via `typing.Tuple` -> tuple has become generic itself, which means it now supports the `[]` syntax. Same for the collection-callable, which has become generic, too. So I think we should just follow along the latest language updates ...

Will close and merge.